core/lib/spree/core/engine.rb

```diff
@@ -68,11 +68,22 @@ class Engine < ::Rails::Engine
 
       # Load in mailer previews for apps to use in development.
       initializer "spree.core.action_mailer.set_preview_path", after: "action_mailer.set_configs" do |app|
-        original_preview_path = app.config.action_mailer.preview_path
-        solidus_preview_path = Spree::Core::Engine.root.join 'lib/spree/mailer_previews'
+        if Rails.gem_version < Gem::Version.new('7.1.0')
+          original_preview_path = app.config.action_mailer.preview_path
+          solidus_preview_path = Spree::Core::Engine.root.join 'lib/spree/mailer_previews'
+
+          app.config.action_mailer.preview_path = "{#{original_preview_path},#{solidus_preview_path}}"
+          ActionMailer::Base.preview_path = app.config.action_mailer.preview_path
+        else
+          app.config.action_mailer.preview_paths << solidus_preview_path.to_s
```

Comment:
`solidus_preview_path` is nil here, as it's defined at line 73 which is inside the other branch of the `if`. When generating the sanbox, this creates an interesting bug by adding the filesystem root to the Rails dependencies autoload path, so when the app loads, it tries to find all rb files on the whole disk, creating 1) huge slowdown, 2) weird file permission access errors (at least on a recent MacOS computer).